libexec/tgenv-upgrade

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -e
+
+[ -n "${TGENV_DEBUG}" ] && set -x
+source "${TGENV_ROOT}/libexec/helpers"
+
+echo "Upgrading tgenv to the latest version..."
+
+if [ -d "$HOME/.tgenv" ]; then
+  cd "$HOME/.tgenv"
+  git checkout main
```

Comment:
I would suggest doing a `git stash` before the checkout.

If the user has made any changes that conflict then `git checkout` will fail I think.

Reply:
✅ done.